Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -135,17 +166,41 @@ class DirectNavigator {
   void initialize(propagator_state_t& state,
                   const stepper_t& /*stepper*/) const {
     ACTS_VERBOSE("Initialize. Surface sequence for navigation:");
-    for (auto surface : state.navigation.options.surfaces) {
+    for (const Surface* surface : state.navigation.options.surfaces) {
       ACTS_VERBOSE(surface->geometryId()
                    << " - " << surface->center(state.geoContext).transpose());
     }
 
     // We set the current surface to the start surface
     state.navigation.currentSurface = state.navigation.options.startSurface;
-    if (state.navigation.currentSurface) {
+    if (state.navigation.currentSurface != nullptr) {
       ACTS_VERBOSE("Current surface set to start surface "
                    << state.navigation.currentSurface->geometryId());
+    } else {
+      ACTS_ERROR("No start surface provided.");
```

Comment:
Hmm this does error, but continues? What does happen without start surface?

Reply:
I guess I could also throw an exception but I felt like `ACTS_ERROR` is what we usually do in the propagation. I think we will just end up running towards the next surface in the stream

Reply:
Hmm I just wonder if we e.g. initialize should return `Result<void>` instead of `void`, to make this properly fail the propagation. But this might be out of the scope of that PR...

Reply:
Yeah I think we do not really forward any hard errors to the propagator at the moment but this is something we can improve on. I think I can put this check into the `makeState` which would cause the failure to pop up earlier and there an exception seems to make more sense.

Reply:
Yeah I think we should be able to return errors, but this would be a larger refactor indeed. Seems a good solution to check it in the `makeState`. Then we could replace the error case with an `assert`?